scripts/fast-reboot

```diff
@@ -170,26 +170,21 @@ function request_pre_shutdown()
     }
 }
 
-function recover_issu_bank_file_instruction()
+function recover_issu_bank_file()
 {
-    debug "To recover (${ISSU_BANK_FILE}) file, do the following:"
-    debug "$ docker exec -it syncd sx_api_dbg_generate_dump.py"
-    debug "$ docker exec -it syncd cat /tmp/sdkdump | grep 'ISSU Bank'"
-    debug "Command above will print the VALUE of ISSU BANK - 0 or 1, use this VALUE in the next command"
-    debug "$ printf VALUE > /host/warmboot/issu_bank.txt"
+    debug "Recovering the (${ISSU_BANK_FILE}) file"
+    docker exec -it syncd sx_api_dbg_generate_dump.py
+    issu_bank_value=`docker exec -it syncd cat /tmp/sdkdump | grep 'ISSU Bank' | grep -o -E '[0-9]+'`
+    printf $issu_bank_value > /host/warmboot/issu_bank.txt
```

Comment:
There command could fail. For example, if syncd container is stopped, the dump.py will fail.

And in the context of `check_issu_bank_file`, you need to check `$1` and exit conditionally.

Reply:
@qiluo-msft I guess, if `syncd` crashed this is very critical, and there is no possibility to proceed with the `warm-reboot` or restoration of `issu_bank.txt`. Could you please clarify about it - how you see the way to proceed?

About the conditional exit, we have email thread with @yxieca and @stepanblyschak, where we discuss to not stop the `warm-reboot` if `issu_bank.txt` is broken.

Reply:
This is my understanding of old behavior:
Based on $BEFORE_PRE_SHUTDOWN
1. before, if anything unexpected happens, immediately stop the reboot
2. after, the `pre-shutdown` already happened, and there is no other choice if anything unepxected happens.

I see there is `#!/bin/bash -e` at the start of the file, so you change both behavior to exit immediately, which is a behavior change.

---
In reply to: [583106643](https://github.com/Azure/sonic-utilities/pull/1466#discussion_r583106643) [](ancestors = 583106643)

Reply:
-e is applicable to the shutdown preparation stage. after we paused swss, -e was removed and we have to proceed no matter what.

Reply:
At this time, we've already stopped swss. The 2 checks are before and after preshutdwon respectively. But at this point, we have to proceed with reboot.

Reply:
Regarding Qi's concern. I think we did enough check before we reach the point to check ISSU file. So the syncd docker should be running (otherwise swss docker won't be in good state for proceeding).

However, it is a fair question that after preshutdown, would sx_api_dbg_generate_dump.py continue to work?

Reply:
@yxieca Could you explain "-e was removed and we have to proceed no matter what." ? Which line removing it?

---
In reply to: [583862133](https://github.com/Azure/sonic-utilities/pull/1466#discussion_r583862133) [](ancestors = 583862133,583106643)

Reply:
Here: https://github.com/Azure/sonic-utilities/pull/1466/files#diff-ce99c55499028be5fd2d0d3247c051eb453e98725c645c395ed29ef9d9e1ad21R528. After this point, we would have shutdown radv/bgp/swss, we cannot revert these, so the only choice is to finish the reboot. Even it becomes a cold one. Of course we are doing all the we can to keep it warm/fast :-)

Reply:
Thanks, understood.

---
In reply to: [585116888](https://github.com/Azure/sonic-utilities/pull/1466#discussion_r585116888) [](ancestors = 585116888,583862133,583106643)